src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -48,11 +49,33 @@ internal FakeManager(Type fakeObjectType, object proxy)
                                          new CallRuleMetadata { Rule = new DefaultReturnValueRule() }
                                      };
 
-            this.recordedCallsField = new List<ICompletedFakeObjectCall>();
+            this.recordedCallsField = new SynchronizedCollection<ICompletedFakeObjectCall>();
             this.interceptionListeners = new LinkedList<IInterceptionListener>();
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="FakeManager"/> class
+        /// for serialization.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo"/> containing the object data.</param>
+        /// <param name="context">The source for this deserialization.</param>
+        protected FakeManager(SerializationInfo info, StreamingContext context)
+        {
+            this.allUserRulesField = (LinkedList<CallRuleMetadata>)info.GetValue("allUserRulesField", typeof(LinkedList<CallRuleMetadata>));
+            this.postUserRules = (CallRuleMetadata[])info.GetValue("postUserRules", typeof(CallRuleMetadata[]));
+            this.preUserRules = (CallRuleMetadata[])info.GetValue("preUserRules", typeof(CallRuleMetadata[]));
+            this.recordedCallsField = new SynchronizedCollection<ICompletedFakeObjectCall>();
+            var recordedCallsToAdd = (ICompletedFakeObjectCall[])info.GetValue("recordedCallsField", typeof(ICompletedFakeObjectCall[]));
+            foreach (var call in recordedCallsToAdd)
+            {
+                this.recordedCallsField.Add(call);
+            }
+
+            this.interceptionListeners = (LinkedList<IInterceptionListener>)info.GetValue("interceptionListeners", typeof(LinkedList<IInterceptionListener>));
+            this.objectReference = (WeakReference)info.GetValue("objectReference", typeof(WeakReference));
+        }

```

Comment:
This makes me a bit uneasy... how to be sure that it will actually behave the same way? I think it would be better to write our own thread-safe collection, make it serializable, and use it instead of `SynchronizedCollection` (which would also solve the enumeration issue I mentioned earlier)